### DISCUSSION TITLE
Audio/sound on FreeBSD

### DISCUSSION THREAD

**Discussion by grahamperrin** [poetaman/arttime#29] [General]
Spun off from PulseAudio-specific https://github.com/poetaman/arttime/discussions/26#discussioncomment-3802098

[Frequently Asked Questions for FreeBSD 12.X and 13.X | FreeBSD Documentation Portal](https://docs.freebsd.org/en/books/faq/) at a glance I see no obvious question or answer, which might be surprising, however – thanks @sergio-carlavilla – a rewrite of the page (currently a _book_) is planned. 

[Audio on FreeBSD – Quick Guide | FreeBSD Foundation](https://freebsdfoundation.org/freebsd-project/resourcesold/audio-on-freebsd/) (2022-06-06, not visibly dated) there's no mention of OSS, but I suspect that OSS is the simplest answer. 

<https://www.freshports.org/www/firefox/#message> is the easiest way for me to remind myself, yes: 

> - `oss` (always available)

[Sound - FreeBSD Wiki](https://wiki.freebsd.org/Sound)

<https://docs.freebsd.org/en/books/handbook/book/#multimedia> (non-split) | <https://docs.freebsd.org/en/books/handbook/multimedia/> (split)

----

```text
% pkg provides en/books/handbook
Name    : en-freebsd-doc-20220825,1
Comment : Documentation from the FreeBSD Documentation Project
Repo    : FreeBSD
Filename: usr/local/share/doc/freebsd/en/books/handbook/handbook_en.pdf
% pkg search -o freebsd-doc | sort
misc/freebsd-doc-all           Documentation from the FreeBSD Documentation Project
misc/freebsd-doc-bn            Bengali translation of the FreeBSD Documentation Project
misc/freebsd-doc-da            Danish translation of the FreeBSD Documentation Project
misc/freebsd-doc-de            German translation of the FreeBSD Documentation Project
misc/freebsd-doc-el            Greek translation of the FreeBSD Documentation Project
misc/freebsd-doc-en            Documentation from the FreeBSD Documentation Project
misc/freebsd-doc-es            Spanish translation of the FreeBSD Documentation Project
misc/freebsd-doc-fr            French translation of the FreeBSD Documentation Project
misc/freebsd-doc-hu            Hungarian translation of the FreeBSD Documentation Project
misc/freebsd-doc-id            Indonesian translation of the FreeBSD Documentation Project
misc/freebsd-doc-it            Italian translation of the FreeBSD Documentation Project
misc/freebsd-doc-ja            Japanese translation of the FreeBSD Documentation Project
misc/freebsd-doc-ko            Korean translation of the FreeBSD Documentation Project
misc/freebsd-doc-mn            Mongolian translation of the FreeBSD Documentation Project
misc/freebsd-doc-nl            Dutch translation of the FreeBSD Documentation Project
misc/freebsd-doc-pl            Polish translation of the FreeBSD Documentation Project
misc/freebsd-doc-pt            Portuguese translation of the FreeBSD Documentation Project
misc/freebsd-doc-ru            Russian translation of the FreeBSD Documentation Project
misc/freebsd-doc-tr            Turkish translation of the FreeBSD Documentation Project
misc/freebsd-doc-zh_cn         Simplified Chinese translation of the FreeBSD Documentation Project
misc/freebsd-doc-zh_tw         Traditional Chinese translation of the FreeBSD Documentation Project
% 
```

@poetaman you might like to install [misc/freebsd-doc-en](https://www.freshports.org/misc/freebsd-doc-en/), then the local PDF of the FreeBSD Handbook can be a useful point of reference: 

`/usr/local/share/doc/freebsd/en/books/handbook/handbook_en.pdf`

**Comment by poetaman** [comment]
@grahamperrin First some background note: I use FreeBSD on Apple Silicon macOS using parallels. I installed X with XFCE and it seems to load well... But Parallel's tools that integrate Mac's mouse/keyboard with DE of guest OS doesn't seem to work for FreeBSD, maybe they haven't ported it for aarch64. So I cannot test this out as usual...

Seems like the answer is ``ogg123`` to play ``.oga`` files. I installed it using home-brew on macOS (``brew install vorbis-tools
``) and works like charm to play ``message-new-instant.oga`` (the freedesktop sound file used by arttime on Linux and BSD). 

Here's how to install it on FreeBSD, its part of ``verbs-tools`` package that seems maintained: .

> https://www.freshports.org/audio/vorbis-tools/
pkg install audio/vorbis-tools
pkg install vorbis-tools

Could you try installing that and running (hope I got the path right on your system):

```
ogg123 /usr/local/share/sounds/freedesktop/stereo/message-new-instant.oga
```

Some testing on Linux VM running on macOS makes ogg123 sound stutter a bit when ogg123 plays the sound for the first time after sounds were played by paplay. So maybe there is an interference between paplay and ogg123 which makes mixing the two commands less ideal. paplay is always clear on Linux VM. I guess the logic should be something like this (not sure if its possible yet): if pulse-audio daemon is already running, then use paplay or if not then if ogg123 command is available then use ogg123 or use paplay.

If you hear the sound well, we can perhaps make ``vorbis-tools`` as a dependency for sounds instead of ``pulse-audio`` on FreeBSD. What do you think? Would it make sense to make ``vorbis-tools`` the default dependency instead of ``pulse-audio`` for all BSDs? On aarch64 macOS it's installation size is a petite 657.5KB.

Thanks for the pointers, might be useful in future!!

**Reply by grahamperrin** [reply]
The package was already installed, sound is audible: 

`ogg123 /usr/local/share/sounds/freedesktop/stereo/message-new-instant.oga`

A test on a different computer might be a good idea. (Not sure about audio in VirtualBox on a FreeBSD host.)

**Comment by poetaman** [comment]
An update with commit and comment is mentioned here: https://github.com/poetaman/arttime/discussions/26#discussioncomment-3830868

**Comment by poetaman** [comment]
Now [PipeWire](https://pipewire.org) is also supported since https://github.com/poetaman/arttime/commit/b4ff7e857889466070ffce05ab85c6b7a593722e. Not sure how it performs on FreeBSD. It is the default on Fedora, and works well there.
